rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,941 +218,203 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
         }
+        port.unwrap()
     }
 
-    /// This gets the list of link-local neighbors, and then contacts them to get their
-    /// Identity using `neighbor_inquiry` as well as their wireguard tunnel name
-    pub fn get_neighbors(&mut self) -> ResponseFuture<Vec<(LocalIdentity, String, IpAddr)>, Error> {
-        KI.trigger_neighbor_disc(&SETTING.get_network().peer_interfaces)
-            .unwrap();
-        let neighs: Vec<
-            Box<Future<Item = Option<(LocalIdentity, String, IpAddr)>, Error = ()>>,
-        > = KI
-            .get_neighbors()
-            .unwrap()
-            .iter()
-            .map(|&(ip_address, ref dev)| (ip_address.to_string(), Some(dev.clone())))
-            .chain({
-                let mut out = Vec::new();
-                if SETTING.get_network().is_gateway && SETTING.get_network().external_nic.is_some()
-                {
-                    for i in SETTING.get_network().manual_peers.clone() {
-                        out.push((i, SETTING.get_network().external_nic.clone()))
-                    }
-                }
-                out
-            }).filter_map(|(ip_address, dev)| {
-                info!("neighbor at interface {:?}, ip {}", dev, ip_address,);
-                if let Some(dev) = dev.clone() {
-                    // Demorgans equal to "if a neighbor is on the listen interfaces and has a valid
-                    // ip or if it's a manual peer and from the right nic don't filter it"
-                    if !(self.listen_interfaces.contains(&dev)
-                        && ip_address.parse::<IpAddr>().is_ok())
-                        && !(SETTING.get_network().external_nic.is_some()
-                            && SETTING
-                                .get_network()
-                                .external_nic
-                                .clone()
-                                .unwrap()
-                                .contains(&dev)
-                            && SETTING.get_network().manual_peers.contains(&ip_address))
-                    {
-                        info!(
-                            "Filtering neighbor at interface {:?}, ip {}",
-                            dev, ip_address,
+    pub fn neighbor_inquiry_hostname(&mut self, their_hostname: String) -> Result<(), Error> {
+        trace!("Getting tunnel, inq");
+
+        // possible port allocation
+        let our_port = self.get_port();
+
+        let res = Resolver::from_registry()
+            .send(resolver::Resolve::host(their_hostname.clone()))
+            .then(move |res| match res {
+                Ok(Ok(dnsresult)) => {
+                    let port = SETTING.get_network().rita_hello_port;
+                    let url = format!("http://[{}]:{}/hello", their_hostname, port);
+                    info!("Saying hello to: {:?} at ip {:?}", url, dnsresult);
+                    if dnsresult.len() > 0 && SETTING.get_network().is_gateway {
```

Comment:
What if we're not a gateway and the DNS request hangs? Several seconds of wasted time

Reply:
This is thought out exactly right! Big reviewer trap though. The delay is neligible thanks to Actix' and futures' shenanigans